src/libmongoc/doc/mongoc_client_pool_get_timeout_ms.rst

```diff
@@ -0,0 +1,24 @@
+:man_page: mongoc_client_pool_get_timeout_ms
+
+mongoc_client_pool_get_timeout_ms()
```

Comment:
Does `mongoc_client_pool_t` require an API to get/set `timeoutMS`?

This may be naive, but I was expecting `timeoutMS` to be managed like RP/RC/WC. Those are maintained on the pool's topology (which contains a `mongoc_uri_t`) and fed to popped clients via `_mongoc_client_new_from_uri`.

Reply:
+1

Reply:
As far as I can tell the spec is not explicit about how timeoutMS gets set.  It only says "This value MUST be configurable at the level of a MongoClient, MongoDatabase, MongoCollection, or of a single operation."  It does not mention pools, but with the approach that I took here of adding setters and getters, it seemed odd to not also be able to set the timeout on the pool.

It sounds like you would prefer that this option can only be set once initially through the URI, which simplifies the child-tracking-parent-inheritance issue as well.

Reply:
libmongoc is unique among drivers in providing a pool of MongoClients, rather than a connection pool _within_ a MongoClient -- so I doubt we''ll ever see a spec address the API for a pool (apart from something like CMAP, perhaps).

My question about "does `mongo_client_pool_t` require an API" was more about libmongoc's own needs, as I wasn't sure if this was being provided for the potential benefit of users, wrapping drivers (N/A for PHP, but I dunno about C++), etc.

If users haven't complained about RC/RP/WC not being configurable on a pool then I'd err on the side of not introducing other configuration methods like `timeoutMS`.

Reply:
Ok, sounds good!  In my next patch I'll remove the setters for pool and client, but leave them in for database and collection.

Reply:
Is there value in providing client level setters/getters, when `timeoutMS` can be configured from the URI?

I can see an argument about consistency with `mongoc_client_set_read_concern`, `mongoc_client_set_write_concern`, `mongoc_client_set_read_prefs`, but that already has some weirdness with pooled clients hanging onto that configuration.

It might be worth removing the client (not just client pool) as well. I think it is still spec compliant – you can still configure it at a client level with a URI option. But it avoids the shenanigans of pooled clients.

Reply:
If there are already known issues with pooled clients retaining state ([CDRIVER-3537](https://jira.mongodb.org/browse/CDRIVER-3537)), I think it makes sense to hold back on adding more client-level setters and instead create a ticket (dependent on CDRIVER-3537) to add that API down the line.

Reply:
All right, I'll remove the client setter as well.  Since the URI functionality is being added in a separate review, I'll just comment out some of the new tests until we can actually set timeout on a client via URI.